tests/karma.js

```diff
@@ -2151,6 +2155,1028 @@ beforeAll(function (done) {
+            'Access-Control-Allow-Headers': accessControlAllowHeaders,
+            'Access-Control-Allow-Methods': 'GET,OPTIONS,POST,PUT,DELETE,MKCOL,PROPFIND,PATCH,PROPPATCH,REPORT'
+          },
+          body: '<?xml version="1.0"?>\n' +
+              '<ocs>\n' +
+              ' <meta>\n' +
+              '  <status>failure</status>\n' +
+              '  <statuscode>404</statuscode>\n' +
+              '  <message>Wrong path, file/folder doesn\'t exist</message>\n' +
+              '  <totalitems></totalitems>\n' +
+              '  <itemsperpage></itemsperpage>\n' +
+              ' </meta>\n' +
+              ' <data/>\n' +
+              '</ocs>'
+        }
+      })
+    }
+    )
+    .then(() => {
+      provider.addInteraction({
+        uponReceiving: 'a GET request for an existent but non-shared file',
+        withRequest: {
+          method: 'GET',
+          path: Pact.Matchers.term({
+            matcher: '.*\\/ocs\\/v1\\.php\\/apps\\/files_sharing\\/api\\/v1\\/shares$',
+            generate: '/ocs/v1.php/apps/files_sharing/api/v1/shares'
+          }),
+          query: 'path=%2FnewFileCreated123',
+          headers: {
+            authorization: 'Basic YWRtaW46YWRtaW4=',
+            Origin: origin
+          }
+        },
+        willRespondWith: {
+          status: 200,
+          headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Headers': accessControlAllowHeaders,
+            'Access-Control-Allow-Methods': 'GET,OPTIONS,POST,PUT,DELETE,MKCOL,PROPFIND,PATCH,PROPPATCH,REPORT'
+          },
+          body: '<?xml version="1.0"?>\n' +
+            '<ocs>\n' +
+            ' <meta>\n' +
+            '  <status>ok</status>\n' +
+            '  <statuscode>100</statuscode>\n' +
+            '  <message/>\n' +
+            '  <totalitems></totalitems>\n' +
+            '  <itemsperpage></itemsperpage>\n' +
+            ' </meta>\n' +
+            ' <data/>\n' +
+            '</ocs>'
+        }
+      })
+    }
+    )
+    .then(() => {
+      provider.addInteraction({
+        uponReceiving: 'a GET request for a non-existent share',
+        withRequest: {
+          method: 'GET',
+          path: Pact.Matchers.term({
+            matcher: '.*\\/ocs\\/v1\\.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/-1$',
+            generate: '/ocs/v1.php/apps/files_sharing/api/v1/shares/-1'
+          }),
+          headers: validAuthHeaders
+        },
+        willRespondWith: {
+          status: 200,
+          headers: xmlResponseHeaders,
+          body: '<?xml version="1.0"?>\n' +
+            '<ocs>\n' +
+            ' <meta>\n' +
+            '  <status>failure</status>\n' +
+            '  <statuscode>404</statuscode>\n' +
+            '  <message>Wrong share ID, share doesn\'t exist</message>\n' +
+            '  <totalitems></totalitems>\n' +
+            '  <itemsperpage></itemsperpage>\n' +
+            ' </meta>\n' +
+            ' <data/>\n' +
+            '</ocs>'
```

Comment:
useful if saved is some variable

Reply:
may be we can fix that in the refactoring